samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -1,276 +1,215 @@
+        SQS_QUEUE_REGEX = rf"^arn:{partition_regex}:sqs:{region_regex}:{account_regex}:[\w-]+$"
+        S3_BUCKET_REGEX = rf"^arn:{partition_regex}:s3:::[\w-]+$"
+        DYNAMODB_TABLE_REGEX = rf"^arn:{partition_regex}:dynamodb:{region_regex}:{account_regex}:table\/[\w-]+$"
+        STEPFUNCTION_REGEX = rf"^arn:{partition_regex}:states:{region_regex}:{account_regex}:stateMachine:[\w-]+$"
+
+        default_permissions_map = {
+            LAMBDA_FUNCTION_REGEX: ["lambda:InvokeFunction"],
+            APIGW_API_REGEX: ["execute-api:Invoke"],
+            APIGW_RESTAPI_REGEX: ["execute-api:Invoke"],
+            SQS_QUEUE_REGEX: ["sqs:SendMessage"],
+            S3_BUCKET_REGEX: ["s3:PutObject", "s3:GetObject"],
+            DYNAMODB_TABLE_REGEX: [
+                "dynamodb:GetItem",
+                "dynamodb:PutItem",
+            ],
+            STEPFUNCTION_REGEX: [
+                "stepfunction:StartExecution",
+                "stepfunction:StopExecution",
+            ],
+        }
+
+        new_statement_template = (
+            "- Effect: Allow\n"
+            "  Action:\n"
+            "   {%- for action in action_list %}\n"
+            "     # - {{action}}\n"
+            "   {%- endfor %}\n"
+            "  Resource: {{arn}}\n"
         )
 
-    if len(resource_tag_mapping_list) == 0:
-        raise KeyError(f"No resources match the tag: {tag_filters}, cannot generate any IAM permissions.")
-
-    iam_statements = [_create_iam_statment_string(resource["ResourceARN"]) for resource in resource_tag_mapping_list]
-    return "".join(iam_statements)
-
-
-def generate_test_runner_template_string(
-    boto_client_provider: BotoProviderType,
-    jinja_base_template: str,
-    s3_bucket_name: str,
-    image_uri: str,
-    tag_filters: Union[dict, None],
-) -> Union[dict, None]:
-    """
-    Renders a base jinja template with a set of parameters needed to create the Test Runner Stack.
-
-    Parameters
-    ----------
-
-    boto_client_provider : BotoProviderType
-        Provides a boto3 client in order to query tagging API
-
-    jinja_base_template : str
-        The jinja YAML template to which parameters are substituted.
-
-    s3_bucket_name : str
-        The name of the S3 bucket used by the Test Runner Stack.
+        for arn_regex, action_list in default_permissions_map.items():
+            if re.search(arn_regex, resource_arn) is not None:
+                LOG.debug("Matched ARN `%s` to IAM actions %s", resource_arn, action_list)
+
+                # APIGW API IAM Statements:
+                # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+                if arn_regex in (APIGW_API_REGEX, APIGW_RESTAPI_REGEX):
+                    apiId = self._extract_api_id_from_arn(resource_arn)
+                    execute_api_arn = (
+                        f"!Sub arn:${{AWS::Partition}}:execute-api:${{AWS::Region}}:${{AWS::AccountId}}:{apiId}/*/GET/*"
```

Comment:
I can't recall; why do we assume GET instead of defaulting to `arn:${{AWS::Partition}}:execute-api:${{AWS::Region}}:${{AWS::AccountId}}:{apiId}/*`?

Reply:
Changed to `/*`, this way any action is authorized from any stage for any resource for a **single api**.